Koha/Schema/Result/InvoiceAdjustment.pm

```diff
@@ -0,0 +1,155 @@
+use utf8;
+package Koha::Schema::Result::InvoiceAdjustment;
+
+# Created by DBIx::Class::Schema::Loader
+# DO NOT MODIFY THE FIRST PART OF THIS FILE
+
+=head1 NAME
+
+Koha::Schema::Result::InvoiceAdjustment
+
+=cut
+
+use strict;
+use warnings;
+
+use base 'DBIx::Class::Core';
+
+=head1 TABLE: C<invoice_adjustments>
+
+=cut
+
+__PACKAGE__->table("invoice_adjustments");
+
+=head1 ACCESSORS
+
+=head2 adjustment_id
+
+  data_type: 'integer'
+  is_auto_increment: 1
+  is_nullable: 0
+
+=head2 invoiceid
+
+  data_type: 'integer'
+  is_foreign_key: 1
+  is_nullable: 0
+
+=head2 adjustment
+
+  data_type: 'decimal'
+  is_nullable: 1
+  size: [28,6]
+
+=head2 reason
+
+  data_type: 'varchar'
+  is_nullable: 1
+  size: 80
+
+=head2 note
+
+  data_type: 'mediumtext'
+  is_nullable: 1
+
+=head2 budget_id
+
+  data_type: 'integer'
+  is_foreign_key: 1
+  is_nullable: 1
+
+=head2 encumber_open
+
+  data_type: 'smallint'
+  default_value: 1
+  is_nullable: 0
+
+=head2 timestamp
+
+  data_type: 'timestamp'
+  datetime_undef_if_invalid: 1
+  default_value: current_timestamp
+  is_nullable: 0
+
+=cut
+
+__PACKAGE__->add_columns(
+  "adjustment_id",
+  { data_type => "integer", is_auto_increment => 1, is_nullable => 0 },
+  "invoiceid",
+  { data_type => "integer", is_foreign_key => 1, is_nullable => 0 },
+  "adjustment",
+  { data_type => "decimal", is_nullable => 1, size => [28, 6] },
+  "reason",
+  { data_type => "varchar", is_nullable => 1, size => 80 },
+  "note",
+  { data_type => "mediumtext", is_nullable => 1 },
+  "budget_id",
+  { data_type => "integer", is_foreign_key => 1, is_nullable => 1 },
+  "encumber_open",
+  { data_type => "smallint", default_value => 1, is_nullable => 0 },
+  "timestamp",
+  {
+    data_type => "timestamp",
+    datetime_undef_if_invalid => 1,
+    default_value => \"current_timestamp",
+    is_nullable => 0,
+  },
+);
+
+=head1 PRIMARY KEY
+
+=over 4
+
+=item * L</adjustment_id>
+
+=back
+
+=cut
+
+__PACKAGE__->set_primary_key("adjustment_id");
+
+=head1 RELATIONS
+
+=head2 budget
+
+Type: belongs_to
+
+Related object: L<Koha::Schema::Result::Aqbudget>
+
+=cut
+
+__PACKAGE__->belongs_to(
+  "budget",
+  "Koha::Schema::Result::Aqbudget",
+  { budget_id => "budget_id" },
+  {
+    is_deferrable => 1,
+    join_type     => "LEFT",
```

Comment:
So you 'can' have an adjustment without a budget to link it to?  That's what `LEFT` here infers.

Reply:
You can, for some reason you can have shipping costs not tied to a budget, so I allowed for this here too. I would assume this is something for internal calculations, we do have some libraries where what they pay in shipping doesn't come from the library budget